src/Symfony/Component/HttpClient/Chunk/ServerSentEvent.php

```diff
@@ -76,4 +78,30 @@ public function getRetry(): float
     {
         return $this->retry;
     }
+
+    /**
+     * Gets the SSE data decoded as an array when it's a JSON payload.
+     */
+    public function getArrayData(): array
+    {
+        if (null !== $this->jsonData) {
+            return $this->jsonData;
+        }
+
+        if ('' === $this->data) {
+            throw new JsonException(sprintf('Server-Sent Event%sdata is empty.', '' !== $this->id ? sprintf(' "%s" ', $this->id) : ' '));
+        }
+
+        try {
+            $jsonData = json_decode($this->data, true, 512, \JSON_BIGINT_AS_STRING | \JSON_THROW_ON_ERROR);
+        } catch (\JsonException $e) {
+            throw new JsonException($e->getMessage().sprintf(' in Server-Sent Event%s.', '' !== $this->id ? sprintf(' "%s"', $this->id) : ''), $e->getCode());
```

Comment:
`throw new JsonException(sprintf('Decoding Server-Sent Event%s failed: ', '' !== $this->id ? sprintf(' "%s"', $this->id) : '').$e->getMessage(), $e->getCode());`